flowmachine/flowmachine/core/server/server.py

```diff
@@ -2,198 +2,55 @@
-            reply = {"status": query_proxy.poll(), "sql": sql}
-
-        elif "get_params" == action:
-            logger.debug(f"Trying to get query parameters. Message: {zmq_msg.msg_str}")
-            query_id = zmq_msg.action_params["query_id"]
-            query_proxy = QueryProxy.from_query_id(query_id)
-            query_run_log.info(
-                "get_params",
-                query_id=query_id,
-                retrieved_params=query_proxy.params,
-                **run_log_dict,
-            )
-            reply = {"id": query_id, "params": query_proxy.params}
-
-        elif "get_query_kind" == action:
-            logger.debug(f"Trying to get query kind. Message: {zmq_msg.msg_str}")
-            query_id = zmq_msg.action_params["query_id"]
-            query_proxy = QueryProxy.from_query_id(query_id)
-            query_run_log.info(
-                "get_query_kind",
-                query_id=query_id,
-                query_kind=query_proxy.query_kind,
-                **run_log_dict,
-            )
-            reply = {"id": query_id, "query_kind": query_proxy.query_kind}
-
-        elif "get_geography" == action:
-            logger.debug(f"Trying to get geography. Message: {zmq_msg.msg_str}")
-            # TODO: Once we have refactored QueryProxy, we won't want to
-            # directly import 'construct_query_object' here.
-            q = construct_query_object("geography", zmq_msg.action_params["params"])
-            # Explicitly project to WGS84 (SRID=4326) to conform with GeoJSON standard
-            sql = q.geojson_query(crs=4326)
-            query_run_log.info("get_geography", **run_log_dict)
-            reply = {"status": QueryState.COMPLETED, "sql": sql}
-
-        else:
-            logger.debug(f"Unknown action: '{action}'")
-            reply = {"status": "error", "error": f"Unknown action: '{action}'"}
-
-    except KeyError as e:
-        reply = {"status": "error", "error": f"Missing key {e}"}
-    except QueryProxyError as e:
-        reply = {"status": "error", "error": f"{e}"}
-    except MissingQueryError as e:
-        reply = {"status": "awol", "id": e.missing_query_id, "error": f"{e}"}
-    except InvalidGeographyError as e:
-        reply = {"status": "awol", "error": f"{e}"}
-    logger.debug(f"Received reply {reply} to message: {zmq_msg.msg_str}")
-    return reply
+        action_name, action_params = parse_zmq_message(msg_str)
+        reply = perform_action(action_name, action_params)
+    except FlowmachineServerError as exc:
+        return ZMQReply(status="error", msg=exc.error_msg).as_json()
 
+    # Return the reply (in JSON format)
+    return reply.as_json()
 
-async def recv(port):
-    """
-    Listen for messages coming in via zeromq on the given port, and dispatch them.
-    """
 
-    ctx = Context.instance()
-    socket = ctx.socket(zmq.ROUTER)
-    socket.bind(f"tcp://*:{port}")
-
-    def shutdown():
```

Comment:
I reckon this is why the coverage has dropped - without this, no coverage is collected for things in flowmachine which get hit in the integration tests.

Reply:
Oh crikey, yes. Need to put that back in too. 🙈